pkg/skaffold/schema/validation/validation_test.go

```diff
@@ -750,3 +752,123 @@ func TestValidateLogsConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateArtifactCircularDependencies(t *testing.T) {
+	tests := []struct {
+		description string
+		artifactLen int
+		dependency  [][]int
+		shouldErr   bool
+	}{
+		{
+			description: "artifacts with no dependency",
+			artifactLen: 5,
+		},
+		{
+			description: "artifacts with dependencies",
+			dependency: [][]int{
+				{1, 3, 4},
+				{3, 2},
+				{2, 4},
```

Comment:
If I understand `setDependency()` correctly, the first index is the artifact index that you are creating dependencies from.  Why are you reversing 3 and 2?